core/block_drag_surface.js

```diff
@@ -139,6 +147,38 @@ Blockly.BlockDragSurfaceSvg.prototype.translateAndScaleGroup = function(x, y, sc
 Blockly.BlockDragSurfaceSvg.prototype.translateSurface = function(x, y) {
   x *= this.scale_;
   y *= this.scale_;
+
+  this.surfaceXY_ = new goog.math.Coordinate(x, y);
+  // This is a work-around to prevent a the blocks from rendering
+  // fuzzy while they are being dragged on the drag surface.
+  x = x.toFixed(0);
+  y = y.toFixed(0);
+  this.SVG_.style.display = 'block';
+  Blockly.utils.setCssTransform(this.SVG_,
+      'translate3d(' + x + 'px, ' + y + 'px, 0px)');
+};
+
+/**
+ * Translate the entire drag surface by the given amount during a drag.
+ * We translate the drag surface instead of the blocks inside the surface
+ * so that the browser avoids repainting the SVG.
+ * Because of this, the drag coordinates must be adjusted by scale.
+ * @param {number} dx X translation delta for the entire surface.
+ * @param {number} dy Y translation delta for the entire surface.
+ * @package
+ */
+Blockly.BlockDragSurfaceSvg.prototype.translateSurfaceBy = function(dx, dy) {
+  goog.asserts.assert(this.surfaceXY_,
+      'Cannot translate block drag surface by a delta when xy_ is null');
+
```

Comment:
it seems like translateSurfaceBy should be able to call translateSurface once it calculates the new position. Then  you don't have to repeat setting surfaceXY, style, etc.